src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c

```diff
@@ -2231,8 +2231,10 @@ char *wm_vuldet_clean_version(const char *version) {
     data = strtok(NULL, ":");
     data = strtok((data) ? data : dup, "-");
 
-    os_calloc(1, strlen(data) + 1, clean_version);
-    strcpy(clean_version, data);
+    if(data) {
+        os_calloc(1, strlen(data) + 1, clean_version);
```

Comment:
This is legacy, but we can improve this by using os_strdup instead of 0s_calloc and strcpy

Reply:
Even more, w_strdup() will also check data nullness

Reply:
You are right, `w_strdup()` is the best choice!